packages/playwright-test/src/util.ts

```diff
@@ -278,3 +278,67 @@ export function getPackageJsonPath(folderPath: string): string {
   return result;
 }
 
+export const levenshteinDistance = (s: string, t: string) => {
+  // See https://en.wikipedia.org/wiki/Levenshtein_distance for reference.
+  // This is a DP implementation variant.
+  const n = s.length;
+  const m = t.length;
+
+  if (n === 0)
+    return m;
+  if (m === 0)
+    return n;
+
+  const d: number[][] = [];
+  // m rows, n cols
+  for (let i = 0; i < m + 1; i++) {
+    const row = [];
+    for (let j = 0; j < n + 1; j++)
+      row.push(0);
+
+    d.push(row);
+  }
+
+  // init row
+  for (let i = 0; i < d[0].length; i++)
+    d[0][i] = i;
+
+  // init column
+  for (let i = 0; i < d.length; i++)
+    d[i][0] = i;
+
+  for (let col = 1; col < n + 1; col++) {
+    for (let row = 1; row < m + 1; row++) {
+      const cost = s[col - 1] === t[row - 1] ? 0 : 1;
```

Comment:
why is it `s[col - 1]` and `t[row - 1]`? the cost of adding a current character should be `s[col] === t[row] ? 0 : 1`.

Reply:
See https://github.com/microsoft/playwright/pull/13229#discussion_r840840489. In the matrix the strings are 1-indexed, but directly accessing them is 0 indexed.